libsolidity/interface/StandardCompiler.cpp

```diff
@@ -472,7 +472,7 @@ std::optional<Json::Value> checkOptimizerDetail(Json::Value const& _details, std
 	return {};
 }
 
-std::optional<Json::Value> checkOptimizerDetailSteps(Json::Value const& _details, std::string const& _name, string& _setting)
+std::optional<Json::Value> checkOptimizerDetailSteps(Json::Value const& _details, std::string const& _name, string& _optimiserSetting, string& _cleanupSetting)
```

Comment:
There's one very important bit missing: the cleanup string is not included in the metadata (see `CompilerStack::createMetadata()`). This is basically the reverse of what we have here. Here we decode and validate the optimizer settings and `createMetadata()` is supposed to serialize them back:
https://github.com/ethereum/solidity/blob/1cfd01c1065ea95468624a28cc57ef9c964bbd31/libsolidity/interface/CompilerStack.cpp#L1538-L1543

Without it being in the metadata, it would now no longer be possible to recreate the same exact bytecode just from metadata. Actually, we really should have some tests trying to do just that but that's out of the scope of this PR. I'll create an issue for that.

Please also grep the repo for `yulOptimiserSteps` and check if there are other places where the cleanup sequence should have been added and was not.

Reply:
In this case, I would assume I'd be adding a new field (to match OptimiserSettings), i.e.
```c++
details["yulDetails"]["optimizerSteps"] = m_optimiserSettings.yulOptimiserCleanupSteps;
```
?

Reply:
As explained by @ekpyron elsewhere, best to just reserialize the sequence back into what the input was.